geomesa-convert/geomesa-convert-xml/src/main/scala/org/locationtech/geomesa/convert/xml/XMLConverter.scala

```diff
@@ -0,0 +1,83 @@
+package org.locationtech.geomesa.convert.xml
+
+import java.io.ByteArrayInputStream
+import javax.xml.parsers.DocumentBuilderFactory
+import javax.xml.xpath.{XPathConstants, XPathExpression, XPathFactory}
+
+import com.google.common.base.Charsets
+import com.typesafe.config.Config
+import org.locationtech.geomesa.convert.Transformers.{EvaluationContext, Expr}
+import org.locationtech.geomesa.convert._
+import org.opengis.feature.simple.SimpleFeatureType
+import org.w3c.dom.NodeList
+
+import scala.collection.JavaConversions._
+
+class XMLConverter(val targetSFT: SimpleFeatureType,
+                   val idBuilder: Expr,
+                   val featurePath: Option[XPathExpression],
+                   val inputFields: IndexedSeq[Field]) extends ToSimpleFeatureConverter[String] {
+
+  private val docBuilder = {
+    val factory = DocumentBuilderFactory.newInstance()
+    factory.setNamespaceAware(false)
+    factory.newDocumentBuilder()
+  }
+
+  override def fromInputType(i: String): Seq[Array[Any]] = {
+    // parse the document once, then extract each feature node and operate on it
+    val root = docBuilder.parse(new ByteArrayInputStream(i.getBytes(Charsets.UTF_8))).getDocumentElement
```

Comment:
Is there a benefit one way or the other? I figured getting the bytes in UTF-8 would be safe, but I'm not entirely sure...